[Security Solution][DQD] Add historical results tour guide

kapral18 opened the pull request:
addresses #195971

This PR adds missing new historical results feature tour guide.

## Tour guide features:
- ability to maintain visual presence while collapsing accordions in list-view
- move from list-view to flyout view and back
- seamlessly integrates with existing opening flyout and history tab functionality

## PR decisions with explanation:
- data-tour-element has been introduced on select elements (like first actions of each first row) to avoid polluting every single element with data-test-subj. This way it's imho specific and semantically more clear what the elements are for.
- early on I tried to control the anchoring with refs but some eui elements don't allow passing refs like EuiTab, so instead a more simpler and straightforward approach with dom selectors has been chosen
- localStorage key name has been picked in accordance with other instances of usage `securitySolution.dataQualityDashboard.historicalResultsTour.v8.16.isActive` the name includes the full domain + the version when it's introduced. And since this tour step is a single step there is no need to stringify an object with `isTourActive` in and it's much simpler to just bake the activity state into the name and make the value just a boolean.

## UI Demo

### Anchor reposition demo (listview + flyout)
https://github.com/user-attachments/assets/0f961c51-0e36-48ca-aab4-bef3b0d1269e

### List view tour guide try it + reload demo
https://github.com/user-attachments/assets/ca1f5fda-ee02-4a48-827c-91df757a8ddf

### FlyOut Try It + reload demo
https://github.com/user-attachments/assets/d0801ac3-1ed1-4e64-9d6b-3140b8402bdf

### Manual history tab selection path + reload demo
https://github.com/user-attachments/assets/34dbb447-2fd6-4dc0-a4f5-682c9c65cc8b

### Manual open history view path + reload demo
https://github.com/user-attachments/assets/945dd042-fc12-476e-8d23-f48c9ded9f65

### Dismiss list view tour guide + reload demo
https://github.com/user-attachments/assets/d20d1416-827f-46f2-9161-a3c0a8cbd932

### Dismiss FlyOut tour guide + reload demo
https://github.com/user-attachments/assets/8f085f59-20a9-49f0-b5b3-959c4719f5cb

### Serverless empty pattern handling + reposition demo
https://github.com/user-attachments/assets/4af5939e-663c-4439-a3fc-deff2d4de7e4



<!--ONMERGE {"backportTargets":["8.x"]} ONMERGE-->

kapral18 committed:
[Security Solution][DQD] Add historical results tour guide

addresses #195971

elasticmachine commented:
Pinging @elastic/security-threat-hunting (Team:Threat Hunting)

elasticmachine commented:
Pinging @elastic/security-threat-hunting-explore (Team:Threat Hunting:Explore)

elasticmachine commented:
## :yellow_heart: Build succeeded, but was flaky
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/242138)
* Commit: f74da261cf606605b85f55d7de5ab33a9883f209

### Failed CI Steps
* [Jest Tests #15](https://buildkite.com/elastic/kibana-pull-request/builds/242138#01928b20-f30d-4e7b-9055-2b551ba65920)
* [FTR Configs #39](https://buildkite.com/elastic/kibana-pull-request/builds/242138#01928b20-f3bc-48a5-a25c-2de221518708)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/242138#01928b20-f3bc-48a5-a25c-2de221518708) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/242138/jobs/01928b20-f3bc-48a5-a25c-2de221518708/artifacts/01928b43-2568-48c2-a21e-19f6c9124dec) FTR Configs #<span></span>39 / Actions and Triggers app create alert should add filter
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/242138#01928b20-f30d-4e7b-9055-2b551ba65920) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/242138/jobs/01928b20-f30d-4e7b-9055-2b551ba65920/artifacts/01928b42-b399-4e5e-9fcc-d77013f0cd17) Jest Tests #<span></span>15 / BadArgument component should display message and help output if command is not hidden from help
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/242138#01928b20-f30d-4e7b-9055-2b551ba65920) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/242138/jobs/01928b20-f30d-4e7b-9055-2b551ba65920/artifacts/01928b42-b398-453d-baac-364d38ccf2ed) Jest Tests #<span></span>15 / BadArgument component should only display message (no help) if command is hidden from help

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/f787b852b23139fbc8e9926263d827ded4a1f451) | [after](https://github.com/elastic/kibana/commit/f74da261cf606605b85f55d7de5ab33a9883f209) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 6001 | 6004 | +3 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/f787b852b23139fbc8e9926263d827ded4a1f451) | [after](https://github.com/elastic/kibana/commit/f74da261cf606605b85f55d7de5ab33a9883f209) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 20.8MB | 20.8MB | +2.7KB |

<details><summary>Unknown metric groups</summary>

#### ESLint disabled line counts
| id | [before](https://github.com/elastic/kibana/commit/f787b852b23139fbc8e9926263d827ded4a1f451) | [after](https://github.com/elastic/kibana/commit/f74da261cf606605b85f55d7de5ab33a9883f209) | diff |
| --- | --- | --- | --- |
| `@kbn/ecs-data-quality-dashboard` | 7 | 8 | +1 |

#### Total ESLint disabled count
| id | [before](https://github.com/elastic/kibana/commit/f787b852b23139fbc8e9926263d827ded4a1f451) | [after](https://github.com/elastic/kibana/commit/f74da261cf606605b85f55d7de5ab33a9883f209) | diff |
| --- | --- | --- | --- |
| `@kbn/ecs-data-quality-dashboard` | 7 | 8 | +1 |


</details>


cc @kapral18

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/242138","number":242138,"commit":"f74da261cf606605b85f55d7de5ab33a9883f209"}],"number":242138}
buildkite-pr-comment-->

angorayc reviewed (commented):
Hey @kapral18 , thanks for adding the tour guide!

Found some cases that we might want to temporarily hide the tour guide:


https://github.com/user-attachments/assets/285a6eb6-a8f8-445e-a703-20663172b4c2

<img width="2560" alt="Screenshot 2024-10-14 at 17 02 48" src="https://github.com/user-attachments/assets/d27c916d-3501-49e8-b5ef-4562fd948027">

semd commented on x-pack/packages/security-solution/ecs_data_quality_dashboard/impl/data_quality_panel/data_quality_details/indices_details/index.tsx:
```diff
@@ -34,19 +35,62 @@ const IndicesDetailsComponent: React.FC<Props> = ({
   const { patternRollups, patternIndexNames } = useResultsRollupContext();
   const { patterns } = useDataQualityContext();
 
+  const [isTourActive, setIsTourActive] = useState<boolean>(() => {
+    const isActive = localStorage.getItem(HISTORICAL_RESULTS_TOUR_IS_ACTIVE_STORAGE_KEY);
+    return isActive !== 'false';
+  });
+
+  const handleDismissTour = useCallback(() => {
+    setIsTourActive(false);
+    localStorage.setItem(HISTORICAL_RESULTS_TOUR_IS_ACTIVE_STORAGE_KEY, 'false');
+  }, []);
```
NIT: Sometimes it's easier to just use `useLocalStorage`. Also, it may be easier to store a _dismissed_ flag instead of _isActive_, this way, when there's no value in the storage it's more obvious to relate that it has not been dismissed.

```
const [isTourDismissed, setIsTourDismissed] = useLocalStorage<boolean>(HISTORICAL_RESULTS_TOUR_IS_DISMISSED_STORAGE_KEY, false);
```

kapral18 replied:
thanks will change

kapral18 replied:
done

semd commented on x-pack/packages/security-solution/ecs_data_quality_dashboard/impl/data_quality_panel/data_quality_details/indices_details/index.tsx:
```diff
@@ -34,19 +35,62 @@ const IndicesDetailsComponent: React.FC<Props> = ({
-          />
-        </StyledPatternWrapperFlexItem>
-      ))}
+      {useMemo(
```
NIT: I don't think this memoization is strictly necessary (I am not saying it's wrong either). But in these cases, I tend to prefer using `React.memo()` in a separate small pure component, than having a `useMemo` that returns React elements. Both options work though, just my five cents.

kapral18 replied:
Yes in most files it's not strictly necessary, but if you add all those strictly unnecessary instances across entire codebase it might become strictly necessary :D But I don't have a proof unless measured, so I agree with you. Especially if now I know there is no specific convention for that. I will remove useMemo's from this file, it will improve on readability a bit.

kapral18 replied:
and with regards to splitting away, I see your point, but this approach can lead to oversplitting and deepening of the render tree. One of the problems that I dealt with during refactoring ecs was exactly that, the unnecessarily splitting of components for the sake of purity :| I flattened them and it's become arguably easier to work with but it cost a gigantic effort of moving folders and imports around and merging tests etc... So it can be a double edged is what I mean, there is always a big unobvious cost.

kapral18 replied:
Done

semd commented on x-pack/packages/security-solution/ecs_data_quality_dashboard/impl/data_quality_panel/data_quality_details/indices_details/pattern/index.tsx:
```diff
@@ -320,6 +358,7 @@ const PatternComponent: React.FC<Props> = ({
                   onCheckNowAction={handleFlyoutCheckNowAndExpandAction}
                   onViewHistoryAction={handleFlyoutViewCheckHistoryAction}
                   sorting={sorting}
+                  firstIndexName={items[0]?.indexName}
```
I see we are already passing `items`, do we really need this new prop? Could we just do `const firstIndexName = items[0]?.indexName` inside the _SummaryTable_ component?

kapral18 replied:
absolutely! good catch, thanks

kapral18 commented:
> Hey @kapral18 , thanks for adding the tour guide!
> 
> Found some cases that we might want to temporarily hide the tour guide:
> 
>  Screen.Recording.2024-10-14.at.17.02.03.mov 
> <img alt="Screenshot 2024-10-14 at 17 02 48" width="2000" src="https://private-user-images.githubusercontent.com/6295984/376315374-d27c916d-3501-49e8-b5ef-4562fd948027.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjg5MjU4NzIsIm5iZiI6MTcyODkyNTU3MiwicGF0aCI6Ii82Mjk1OTg0LzM3NjMxNTM3NC1kMjdjOTE2ZC0zNTAxLTQ5ZTgtYjVlZi00NTYyZmQ5NDgwMjcucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTAxNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEwMTRUMTcwNjEyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YmMyMDc4MzAwZjI1YWYwNGUxZTI2Mzc5OWM1YmZhYTdjODgzNzFhMjc3ODExYjA1ZDdjOWFhNGZjNDJhNWU2MyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.Yft36F-96yVFFYkT7i6yDi7JIjsowjCvlgfUv7nOeWQ">

I'm so glad I asked you to test :) I completely forgot about timeline. thank you!

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/242385)
* Commit: e92a06fcc874390befea393416da7f9865c8b1df

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/f787b852b23139fbc8e9926263d827ded4a1f451) | [after](https://github.com/elastic/kibana/commit/e92a06fcc874390befea393416da7f9865c8b1df) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 6001 | 6004 | +3 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/f787b852b23139fbc8e9926263d827ded4a1f451) | [after](https://github.com/elastic/kibana/commit/e92a06fcc874390befea393416da7f9865c8b1df) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 20.8MB | 20.8MB | +2.7KB |

<details><summary>Unknown metric groups</summary>

#### ESLint disabled line counts
| id | [before](https://github.com/elastic/kibana/commit/f787b852b23139fbc8e9926263d827ded4a1f451) | [after](https://github.com/elastic/kibana/commit/e92a06fcc874390befea393416da7f9865c8b1df) | diff |
| --- | --- | --- | --- |
| `@kbn/ecs-data-quality-dashboard` | 7 | 8 | +1 |

#### Total ESLint disabled count
| id | [before](https://github.com/elastic/kibana/commit/f787b852b23139fbc8e9926263d827ded4a1f451) | [after](https://github.com/elastic/kibana/commit/e92a06fcc874390befea393416da7f9865c8b1df) | diff |
| --- | --- | --- | --- |
| `@kbn/ecs-data-quality-dashboard` | 7 | 8 | +1 |


</details>


### History
* :yellow_heart: [Build #242138](https://buildkite.com/elastic/kibana-pull-request/builds/242138) was flaky f74da261cf606605b85f55d7de5ab33a9883f209

cc @kapral18

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/242385","number":242385,"commit":"e92a06fcc874390befea393416da7f9865c8b1df"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/242138","number":242138,"commit":"f74da261cf606605b85f55d7de5ab33a9883f209"}],"number":242385}
buildkite-pr-comment-->

kapral18 committed:
fix: fix edge cases

- fix tour guide's visibility of empty patterns (happens mostly in
  serverless)
- fix eui bug preventing from proper recalculation of tour guide popover
  position on surrounding accordions' toggle
- reduce potential flakiness of some tests in CI with increased timeouts

kapral18 committed:
chore: address comments

- Renamed storage key for historical results tour from
`HISTORICAL_RESULTS_TOUR_IS_ACTIVE_STORAGE_KEY` to
`HISTORICAL_RESULTS_TOUR_IS_DISMISSED_STORAGE_KEY`.
- Added a new custom hook `useIsHistoricalResultsTourActive` to manage
the state of the historical results tour.
- Updated `IndicesDetails` component to use the new hook for determining
if the tour is active.
- Modified tests to reflect changes in storage key and tour dismissal
logic.
- Adjusted `HistoricalResultsTour` component to accept an optional
`zIndex` prop for better positioning.
- removed unnecessary useMemo hooks

angorayc commented:
@kapral18 , 
I found the tour is cropped when opening the flyout. Could you please try if it is reproducible?

 
https://github.com/user-attachments/assets/ce8f3ea9-5770-4579-8abd-9b9308350d1b

semd reviewed (approved):
LGTM! Thanks for taking the time to address the comments.

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/242635)
* Commit: 78906bdb7bf709c0b02ec7142178faa8fcef8d20

### Failed CI Steps
* [Jest Tests #1](https://buildkite.com/elastic/kibana-pull-request/builds/242635#01929086-f6a3-4fe7-9628-b8b1a74fcfca)
* [Jest Tests #1](https://buildkite.com/elastic/kibana-pull-request/builds/242635#019290b3-53a3-4c0a-b11b-cff76e27b49d)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/242635#01929086-f6a3-4fe7-9628-b8b1a74fcfca) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/242635/jobs/01929086-f6a3-4fe7-9628-b8b1a74fcfca/artifacts/019290b3-0fc5-407c-ae08-3170dd982638) Jest Tests #<span></span>1 / IndicesDetails tour when the tour is dismissed it hides the tour and persists in localStorage
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/242635#019290b3-53a3-4c0a-b11b-cff76e27b49d) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/242635/jobs/019290b3-53a3-4c0a-b11b-cff76e27b49d/artifacts/019290bc-27c9-4296-8e6b-bba63f8a92d4) Jest Tests #<span></span>1 / IndicesDetails tour when the tour is dismissed it hides the tour and persists in localStorage

### History
* :green_heart: [Build #242385](https://buildkite.com/elastic/kibana-pull-request/builds/242385) succeeded e92a06fcc874390befea393416da7f9865c8b1df
* :yellow_heart: [Build #242138](https://buildkite.com/elastic/kibana-pull-request/builds/242138) was flaky f74da261cf606605b85f55d7de5ab33a9883f209

cc @kapral18

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/242635","number":242635,"commit":"78906bdb7bf709c0b02ec7142178faa8fcef8d20"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/242385","number":242385,"commit":"e92a06fcc874390befea393416da7f9865c8b1df"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/242138","number":242138,"commit":"f74da261cf606605b85f55d7de5ab33a9883f209"}],"number":242635}
buildkite-pr-comment-->

kapral18 committed:
chore(tests): fix trailing test

kapral18 commented:
> @kapral18 , I found the tour is cropped when opening the flyout. Could you please try if it is reproducible?
> 
>  Screen.Recording.2024-10-15.at.15.21.41.mov

thanks for checking but cannot reproduce this

https://github.com/user-attachments/assets/a8bf245b-f234-4365-b26e-f63123f7f6da

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/242714)
* Commit: 7744ee914da33d56295fd4dfd0681e38cdcc6e9c

### Failed CI Steps
* [Linting](https://buildkite.com/elastic/kibana-pull-request/builds/242714#019290d0-3be9-4401-9595-2ececcbc480c)

### History
* :broken_heart: [Build #242635](https://buildkite.com/elastic/kibana-pull-request/builds/242635) failed 78906bdb7bf709c0b02ec7142178faa8fcef8d20
* :green_heart: [Build #242385](https://buildkite.com/elastic/kibana-pull-request/builds/242385) succeeded e92a06fcc874390befea393416da7f9865c8b1df
* :yellow_heart: [Build #242138](https://buildkite.com/elastic/kibana-pull-request/builds/242138) was flaky f74da261cf606605b85f55d7de5ab33a9883f209

cc @kapral18

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/242714","number":242714,"commit":"7744ee914da33d56295fd4dfd0681e38cdcc6e9c"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/242635","number":242635,"commit":"78906bdb7bf709c0b02ec7142178faa8fcef8d20"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/242385","number":242385,"commit":"e92a06fcc874390befea393416da7f9865c8b1df"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/242138","number":242138,"commit":"f74da261cf606605b85f55d7de5ab33a9883f209"}],"number":242714}
buildkite-pr-comment-->

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/242797)
* Commit: ab2dede3d96cf5a4ad449660f3e42746dd1f98ff

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/242797#01929164-3bdb-418e-a53c-7956af41eee0)

### History
* :broken_heart: [Build #242714](https://buildkite.com/elastic/kibana-pull-request/builds/242714) failed 7744ee914da33d56295fd4dfd0681e38cdcc6e9c
* :broken_heart: [Build #242635](https://buildkite.com/elastic/kibana-pull-request/builds/242635) failed 78906bdb7bf709c0b02ec7142178faa8fcef8d20
* :green_heart: [Build #242385](https://buildkite.com/elastic/kibana-pull-request/builds/242385) succeeded e92a06fcc874390befea393416da7f9865c8b1df
* :yellow_heart: [Build #242138](https://buildkite.com/elastic/kibana-pull-request/builds/242138) was flaky f74da261cf606605b85f55d7de5ab33a9883f209

cc @kapral18

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/242797","number":242797,"commit":"ab2dede3d96cf5a4ad449660f3e42746dd1f98ff"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/242714","number":242714,"commit":"7744ee914da33d56295fd4dfd0681e38cdcc6e9c"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/242635","number":242635,"commit":"78906bdb7bf709c0b02ec7142178faa8fcef8d20"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/242385","number":242385,"commit":"e92a06fcc874390befea393416da7f9865c8b1df"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/242138","number":242138,"commit":"f74da261cf606605b85f55d7de5ab33a9883f209"}],"number":242797}
buildkite-pr-comment-->

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/242820)
* Commit: 06c7bbb4a0ee325e129dcb5eafb1f9c53f984bc0

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/242820#0192919f-877e-472e-933a-8783b2f38f57)

### History
* :broken_heart: [Build #242797](https://buildkite.com/elastic/kibana-pull-request/builds/242797) failed ab2dede3d96cf5a4ad449660f3e42746dd1f98ff
* :broken_heart: [Build #242714](https://buildkite.com/elastic/kibana-pull-request/builds/242714) failed 7744ee914da33d56295fd4dfd0681e38cdcc6e9c
* :broken_heart: [Build #242635](https://buildkite.com/elastic/kibana-pull-request/builds/242635) failed 78906bdb7bf709c0b02ec7142178faa8fcef8d20
* :green_heart: [Build #242385](https://buildkite.com/elastic/kibana-pull-request/builds/242385) succeeded e92a06fcc874390befea393416da7f9865c8b1df
* :yellow_heart: [Build #242138](https://buildkite.com/elastic/kibana-pull-request/builds/242138) was flaky f74da261cf606605b85f55d7de5ab33a9883f209

cc @kapral18

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/242820","number":242820,"commit":"06c7bbb4a0ee325e129dcb5eafb1f9c53f984bc0"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/242797","number":242797,"commit":"ab2dede3d96cf5a4ad449660f3e42746dd1f98ff"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/242714","number":242714,"commit":"7744ee914da33d56295fd4dfd0681e38cdcc6e9c"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/242635","number":242635,"commit":"78906bdb7bf709c0b02ec7142178faa8fcef8d20"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/242385","number":242385,"commit":"e92a06fcc874390befea393416da7f9865c8b1df"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/242138","number":242138,"commit":"f74da261cf606605b85f55d7de5ab33a9883f209"}],"number":242820}
buildkite-pr-comment-->

elasticmachine commented:
## :yellow_heart: Build succeeded, but was flaky
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/242881)
* Commit: 0d85881a90b2bdfa4ad2148919d0cac8564c785c

### Failed CI Steps
* [FTR Configs #42](https://buildkite.com/elastic/kibana-pull-request/builds/242881#0192922c-22a0-4555-8141-710d70c27f01)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/242881#0192922c-22a0-4555-8141-710d70c27f01) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/242881/jobs/0192922c-22a0-4555-8141-710d70c27f01/artifacts/01929253-df38-4d7e-96d7-b063f5126385) FTR Configs #<span></span>42 / Actions and Triggers app Maintenance Windows Maintenance window update form should update a maintenance window

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/da63469091f9b76e9e5d053cf11a4df15e14fdef) | [after](https://github.com/elastic/kibana/commit/0d85881a90b2bdfa4ad2148919d0cac8564c785c) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 6016 | 6020 | +4 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/da63469091f9b76e9e5d053cf11a4df15e14fdef) | [after](https://github.com/elastic/kibana/commit/0d85881a90b2bdfa4ad2148919d0cac8564c785c) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 20.6MB | 20.6MB | +3.0KB |

<details><summary>Unknown metric groups</summary>

#### ESLint disabled line counts
| id | [before](https://github.com/elastic/kibana/commit/da63469091f9b76e9e5d053cf11a4df15e14fdef) | [after](https://github.com/elastic/kibana/commit/0d85881a90b2bdfa4ad2148919d0cac8564c785c) | diff |
| --- | --- | --- | --- |
| `@kbn/ecs-data-quality-dashboard` | 7 | 8 | +1 |

#### Total ESLint disabled count
| id | [before](https://github.com/elastic/kibana/commit/da63469091f9b76e9e5d053cf11a4df15e14fdef) | [after](https://github.com/elastic/kibana/commit/0d85881a90b2bdfa4ad2148919d0cac8564c785c) | diff |
| --- | --- | --- | --- |
| `@kbn/ecs-data-quality-dashboard` | 7 | 8 | +1 |


</details>


### History
* :broken_heart: [Build #242820](https://buildkite.com/elastic/kibana-pull-request/builds/242820) failed 06c7bbb4a0ee325e129dcb5eafb1f9c53f984bc0
* :broken_heart: [Build #242797](https://buildkite.com/elastic/kibana-pull-request/builds/242797) failed ab2dede3d96cf5a4ad449660f3e42746dd1f98ff
* :broken_heart: [Build #242714](https://buildkite.com/elastic/kibana-pull-request/builds/242714) failed 7744ee914da33d56295fd4dfd0681e38cdcc6e9c
* :broken_heart: [Build #242635](https://buildkite.com/elastic/kibana-pull-request/builds/242635) failed 78906bdb7bf709c0b02ec7142178faa8fcef8d20
* :green_heart: [Build #242385](https://buildkite.com/elastic/kibana-pull-request/builds/242385) succeeded e92a06fcc874390befea393416da7f9865c8b1df

cc @kapral18

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/242881","number":242881,"commit":"0d85881a90b2bdfa4ad2148919d0cac8564c785c"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/242820","number":242820,"commit":"06c7bbb4a0ee325e129dcb5eafb1f9c53f984bc0"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/242797","number":242797,"commit":"ab2dede3d96cf5a4ad449660f3e42746dd1f98ff"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/242714","number":242714,"commit":"7744ee914da33d56295fd4dfd0681e38cdcc6e9c"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/242635","number":242635,"commit":"78906bdb7bf709c0b02ec7142178faa8fcef8d20"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/242385","number":242385,"commit":"e92a06fcc874390befea393416da7f9865c8b1df"}],"number":242881}
buildkite-pr-comment-->

kibanamachine commented:
Starting backport for target branches: 8.x

https://github.com/elastic/kibana/actions/runs/11355862524

kibanamachine commented:
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.x|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/196456">](https://github.com/elastic/kibana/pull/196456)|

Note: Successful backport PRs will be merged automatically after passing CI.

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

<!--- Backport version: 9.4.3 -->
